pkg/api/pluginproxy/access_token_provider.go

```diff
@@ -44,9 +44,41 @@ type accessTokenProvider struct {
 }
 
 type jwtToken struct {
-	ExpiresOn       time.Time `json:"-"`
-	ExpiresOnString string    `json:"expires_on"`
-	AccessToken     string    `json:"access_token"`
+	ExpiresOn   time.Time
+	AccessToken string
+}
+
+func (token *jwtToken) UnmarshalJSON(b []byte) error {
+	var t struct {
+		AccessToken string      `json:"access_token"`
+		ExpiresOn   json.Number `json:"expires_on"`
+		ExpiresIn   json.Number `json:"expires_in"`
+	}
+
+	if err := json.Unmarshal(b, &t); err != nil {
+		return err
+	}
+
+	token.AccessToken = t.AccessToken
+
+	switch {
+	case len(t.ExpiresOn) != 0:
+		expiresOn, err := t.ExpiresOn.Int64()
+		if err != nil {
+			return err
+		}
+		token.ExpiresOn = time.Unix(expiresOn, 0)
+	case len(t.ExpiresIn) != 0:
+		expiresIn, err := t.ExpiresIn.Int64()
+		if err != nil {
+			return err
+		}
+		token.ExpiresOn = time.Unix(time.Now().Unix()+(expiresIn), 0)
```

Comment:
You could rewrite this line so as to be more readable like:
`token.ExpiresOn = time.Now().Add(time.Duration(expiresIn) * time.Second)`